pkg/apis/clusteroperator/types.go

```diff
@@ -48,6 +48,48 @@ type ClusterList struct {
 	Items []Cluster
 }
 
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ClusterVersion represents a version of OpenShift that can be, or is installed.
+type ClusterVersion struct {
+	// +optional
+	metav1.TypeMeta
+	// +optional
+	metav1.ObjectMeta
+
+	// Possible future additions: RPMRepositories, openshift_release / image_tag / pkg_version.
+
+	// ImageFormat defines the pattern string for finding required OpenShift images.
+	// (i.e. "openshift/origin-${component}:v3.7.9", or
+	// "ci.openshift.org/ci-jobs/pr-2034834098243:${component}")
+	// Converts to oreg_url for openshift-ansible.
+	ImageFormat string
```

Comment:
I'm thinking that we should follow convention and have a Spec and Status structure. At some point we may want to have a controller that works on version objects to auto-discover things like AMI names.

Reply:
Ok good idea, I will add.